slave/url_creator.py

```diff
@@ -64,21 +64,53 @@ def latest(self, **kwargs):
 class TaskClusterIndexHelper(object):
     _index_url = 'https://index.taskcluster.net/v1/task/gecko.v2'
     _artifacts = 'https://public-artifacts.taskcluster.net'
+    _task_inspector = 'https://tools.taskcluster.net/tasks'
+
+    BUILDTYPES = ('debug', 'nightly', 'opt', 'pgo')
+    PRODUCTS = ('firefox', 'mobile')
+    PLATFORMS = ('linux', 'linux64', 'macosx64', 'win32', 'win64')
 
     @classmethod
-    def _latest_unsigned_nightly_task_id(cls, repo_name, product, platform):
-        '''Return taskId for the latest nightly task.'''
-        assert product in ('firefox', 'mobile')
-        # Even though it says 'opt' these are PGO nightly builds
-        platform = platform + '-opt'
-        assert platform in ('linux-opt', 'linux64-opt', 'macosx64-opt',
-                            'win32-opt', 'win64-opt')
-
-        url = '{}.{}.nightly.latest.{}.{}'.format(
-            cls._index_url,
-            repo_name,
-            product,
-            platform)
+    def build_url(cls, repo_name, product, platform, buildtype, revision=None):
+        task_id = cls._task_id(repo_name, product, platform, buildtype, revision)
+        print 'TASK: %s/%s' % (cls._task_inspector, task_id)
```

Comment:
Yes, correct. This can allow someone determine where an artifact was downloaded for.
It's for auditing purposes.